### DISCUSSION TITLE
Nuclei not responding Error When Scanning Free SNI Websites on Free Mode Internet Connection

### DISCUSSION THREAD

**Discussion by 0x4hm3d0x** [projectdiscovery/nuclei-templates#7494] [Q&A] [answered]
Issue Description

I am encountering a timeout error when scanning free SNI (Server Name Indication) websites using the latest version of the nuclei tool on a free mode internet connection. It appears that the tool may be attempting SSH connections or connecting to other proxies that require a data internet connection, leading to the timeout error.

Environment

Nuclei Version: v2.9.6
Am using Termux  emulator on Android 

Execute the following command to scan a free SNI website (e.g., free.facebook.com) on a free mode internet connection:
nuclei -target free.facebook.com

```console
~ $ nuclei -target free.facebook.com

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.9.6

                projectdiscovery.io

[INF] Current nuclei version: v2.9.6 (outdated)
[INF] Current nuclei-templates version: v9.5.2 (latest)
[INF] New templates added in latest release: 50
[INF] Templates loaded for current scan: 6064
[INF] Targets loaded for current scan: 1
[INF] Running httpx on input host
[INF] Found 1 URL from httpx
[INF] Templates clustered: 1075 (Reduced 1015 Requests)
```

I expect the nuclei tool to perform the scan on the specified target (free SNI website) without encountering a timeout error, taking into account the limitations of a free mode internet connection.

Instead of successfully scanning the free SNI website, the nuclei tool throws a timeout error. It seems that the tool is attempting to establish SSH connections or connect to other proxies that require a data internet connection, which is not available in the free mode.

Additional Information

The timeout error consistently occurs when scanning free SNI websites on a free mode internet connection.
I have verified that the issue is not related to the availability of the target website or any network restrictions on my end.
This problem does not manifest when scanning non-SNI websites or when using the nuclei tool on a regular internet connection.

**Comment by ehsandeep** [comment]
@0x4hm3d0x not sure what exactly you are referring by Free SNI Website, but nuclei scan works as usual for me.

```console
nuclei -target free.facebook.com

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.9.6

		projectdiscovery.io

[INF] Current nuclei version: v2.9.6 (latest)
[INF] Current nuclei-templates version: v9.5.2 (latest)
[INF] New templates added in latest release: 50
[INF] Templates loaded for current scan: 6064
[INF] Targets loaded for current scan: 1
[INF] Running httpx on input host
[INF] Found 1 URL from httpx
[INF] Templates clustered: 1077 (Reduced 1017 Requests)
[httponly-cookie-detect] [http] [info] https://free.facebook.com
[ssl-issuer] [ssl] [info] free.facebook.com:443 [DigiCert Inc]
[ssl-dns-names] [ssl] [info] free.facebook.com:443 [*.messenger.com,*.xx.fbcdn.net,*.xy.fbcdn.net,*.xz.fbcdn.net,messenger.com,*.facebook.net,*.m.facebook.com,*.fbsbx.com,facebook.com,*.facebook.com,*.fbcdn.net]
[INF] Using Interactsh Server: oast.fun
[waf-detect:ats] [http] [info] https://free.facebook.com/
[cname-fingerprint] [dns] [info] free.facebook.com [z-m.facebook.com.]
[deprecated-tls] [ssl] [info] free.facebook.com:443 [tls10]
[deprecated-tls] [ssl] [info] free.facebook.com:443 [tls11]
[apple-app-site-association] [http] [info] https://free.facebook.com/.well-known/apple-app-site-association
[tls-version] [ssl] [info] free.facebook.com:443 [tls10]
[tls-version] [ssl] [info] free.facebook.com:443 [tls11]
[tls-version] [ssl] [info] free.facebook.com:443 [tls12]
[tls-version] [ssl] [info] free.facebook.com:443 [tls13]
[weak-cipher-suites] [ssl] [medium] free.facebook.com:443 [[tls10 TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA]]
[weak-cipher-suites] [ssl] [medium] free.facebook.com:443 [[tls11 TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA]]
[robots-txt-endpoint] [http] [info] https://free.facebook.com/robots.txt
[robots-txt-endpoint] [http] [info] https://free.facebook.com/tr
[robots-txt-endpoint] [http] [info] https://free.facebook.com/tr/
[robots-txt-endpoint] [http] [info] https://free.facebook.com/careers/
```

**Comment by 0x4hm3d0x** [comment]
@ehsandeep When I attempt to scan websites on a free mode internet connection, such as websites with SNI (Server Name Indication) like "free.facebook.com" or similar domains, I encounter a timeout error. It seems that Nuclei may be attempting to establish an SSH connection or connect to a proxy server that requires an active internet data connection.

As a result, the tool is not able to perform the scan successfully and remains unresponsive. I've confirmed that this issue occurs specifically when using Nuclei on a free mode internet connection.

**Comment by 0x4hm3d0x** [comment]
@ehsandeep When I attempt to scan websites on a free mode internet connection, such as websites with SNI (Server Name Indication) like "free.facebook.com" or similar domains, I encounter a timeout error. It seems that Nuclei may be attempting to establish an SSH connection or connect to a proxy server that requires an active internet data connection.

As a result, the tool is not able to perform the scan successfully and remains unresponsive. I've confirmed that this issue occurs specifically when using Nuclei on a free mode internet connection.

**Comment by 0x4hm3d0x** [comment]
@ehsandeep A "Free SNI Website" refers to a website that can be accessed without the need for a paid internet data connection. SNI stands for Server Name Indication, which is an extension of the TLS protocol. It allows a client (such as a web browser) to indicate the hostname of the website it wants to access during the TLS handshake process.

In some cases, certain websites may offer free access or have a limited version that can be accessed without the need for a paid internet data plan. These websites typically have a domain or subdomain that includes the term "free" in it, indicating their availability for free access.

For example, "free.facebook.com" would be a Free SNI Website, where users can access a version of Facebook without incurring any data charges. Similarly, there might be other websites that offer similar free access options.

**Comment by ehsandeep** [comment, marked as answer]
@0x4hm3d0x nuclei require a proper network connection, so this won't work.

**Reply by 0x4hm3d0x** [reply]
Aa OK  I understand thanks
